architecture/consistency.md

```diff
@@ -0,0 +1,96 @@
+
+#### Transactions bound to partitions
+
+Some cloud services like Azure Table Storage, Azure Cosmos DB Table API, and Azure Cosmos DB SQL API offer transactions, but they are usually scoped to a single partition key. When transactions are scoped to a single partition key, all storage operations that need to be atomic have to share the same partition key on all read, update and delete operations to achieve atomicity.
+
+#### No distributed transactions
+
+Distributed transactions (transactions that span multiple services, e.g. a database and a message queue) are required to ensure strong consistency of all operations within the shared transaction. On Windows, the Distributed Transaction Coordinator (DTC) is able coordinate transactions across multiple compatible participants using two-phase commit protocols. Each participating resource of a distributed transaction must explicitly support the two-phase commit protocol. While traditional on-premises focused services like MSMQ or MSSQL server support the DTC, managed cloud services do not support DTC transactions, enforcing different strategies to achieve consistency across transactional resources.
+
+### Outbox Pattern
+
+The [transactional Outbox Pattern](https://microservices.io/patterns/data/transactional-outbox.html) is a pattern to achieve atomically consitent database and message queueing operations. Due to the lack of distributed transactions in many modern application environments, the outbox pattern is a commonly used pattern to solve strong consistency requirements.
+
+The outbox pattern is implemented by storing outgoing messages into the same database as the business operations. This allows the use of database native transaction capabilities to achieve atomic consistency. In a second step, the persisted messages are dispatched to tbe message queue.
+
+The nature of the outbox pattern might cause outgoing messages to be dispatched multiple times. This should be taken into consideration by leveraging [idempotency mechanisms](#idempotency) on the receiver.
+
+Note: Implementing the outbox pattern is very risky and error-prone. Small mistakes can lead to unintended behavior and message or data loss. Consider the the [NServiceBus Outbox feature](/nservicebus/outbox/) which implements the Outbox pattern, including built-in message deduplication when receiving messages. The NServiceBus Outbox is used in production by hundreds of customers, thoroughly tested and well documented.
+
+
+#### Transactional Session
+
+The TransactionalSession is an NServiceBus feature specifically designed to bring Outbox functionality to system boundaries like web servers that require outgoing message operations and database operations to be consistent.
```

Comment:
```suggestion
```

Thinking more about this, I feel we can remove this since the transactional session is also just an implementation of the outbox pattern. The outbox documentation will point to the tx session too.